virttest/qemu_devices/qdevices.py

```diff
@@ -2236,7 +2236,8 @@ def is_direct_plug(self, device):
                         "vhost-vsock-pci", "virtio-net",
                         "virtio-scsi-pci", "virtio-balloon-pci",
                         "virtio-serial-pci", "virtio-rng-pci",
-                        "e1000e", "virtio-gpu-pci", "qemu-xhci"]
+                        "e1000e", "virtio-gpu-pci", "qemu-xhci",
+                        "VGA"]
```

Comment:
Seems different platforms have different topology representation, with using the following libvirt domain configuration,
```xml
<video>
  <model type='vga'/>
</video>
```
the outcome of qemu command is:
```
# x86_64 -M q35
    -device VGA,id=video0,vgamem_mb=16,bus=pcie.0,addr=0x1 \
```
```
# aarch64 -M virt
    -device pcie-root-port,port=0x8,chassis=1,id=pci.1,bus=pcie.0,multifunction=on,addr=0x1 \
    -device pcie-pci-bridge,id=pci.3,bus=pci.1,addr=0x0 \
    -device VGA,id=video0,vgamem_mb=16,bus=pci.3,addr=0x1 \
```
so, have to distinguish them, thanks.

Reply:
I don't think it's worth changing, let's keep it the q35 way (original).

I assume there used to be a problem putting the VGA to a pcie-root-port before so they added the hook to inject it directly. On arm nobody probably tried that so it uses the default. I guess (don't know for sure) that they will start plugging the VGA to a root-port in the future and then we might add it to the list of direct-plugs.

So let me simply remove this patch for now :smile:

Reply:
ok, sounds good, thanks.